File: [23.12]Design-Pattern/Java/src/behavioral/iterator/Main.java
package behavioral.iterator;

public class Main {
    public static void main(String[] args) {
        HamburgerCollection hamburgerCollection = new HamburgerCollection(5);
        hamburgerCollection.add(new Hamburger("기본 햄버거", 500));
        hamburgerCollection.add(new Hamburger("치즈 햄버거", 1000));
        hamburgerCollection.add(new Hamburger("불고기 햄버거", 1000));
        hamburgerCollection.add(new Hamburger("석쇠 불고기 햄버거", 1500));
        hamburgerCollection.add(new Hamburger("치즈 불고기 햄버거", 1500));

        Iterator iter = hamburgerCollection.iterator();

        while (iter.hasNext()) {
            System.out.println(iter.next());
        }

    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/behavioral/iterator/main.kt
package behavioral.iterator

fun main() {
    val hamburgerCollection = HamburgerCollection(5)
    hamburgerCollection.add(Hamburger(5000, "햄버거"))
    hamburgerCollection.add(Hamburger(10000, "치즈 햄버거"))
    hamburgerCollection.add(Hamburger(10000, "불고기 햄버거"))
    hamburgerCollection.add(Hamburger(15000, "치즈 불고기 햄버거"))
    hamburgerCollection.add(Hamburger(10000, "치킨 햄버거"))

    val iter = hamburgerCollection.iterator()

    while (iter.hasNext()) {
        println(iter.next())
    }

}